### PR TITLE
added modeling rule

### PR DESCRIPTION
**Pull request by sharonfi99**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: [link to the issue](https://jira-dc.paloaltonetworks.com/browse/CIAC-11209)

## Description
Adding modeling rule for IBM Security Verify.
## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

**Review by OBavly — APPROVED** [review]

**Comment by ShirleyDenkberg** [comment]
@OBavly Doc review completed.

File: Packs/IBMSecurityVerify/ModelingRules/IBMSecurityVerify/IBMSecurityVerify.xif [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+    xdm.source.user.groups = to_json_string(data -> entitlements) -> []
+| call generic_fields_mapping;
+
+/*
+Mapping of 'sso' event types.
+*/
+filter event_type = "sso"
+| alter
+    src_user = arrayindex(regextract(data -> username, "(.+)\@") ,0) // The username can sometimes appear as a username and other times as an email.
+| alter
```
Comment (yassorpanw) [inline Packs/IBMSecurityVerify/ModelingRules/IBMSecurityVerify/IBMSecurityVerify.xif:56]: For the the relevant events for 102 story, in our case it's "**sso**" and "**authentication**" events,  we need to verify the following:
1. xdm.event.type = "authentication"
2. xdm.event.tags = arraycreate(XDM_CONST.EVENT_TAG_AUTHENTICATION)
3. Have 5 tuples (even zero if not exists)
4. All of the following fields contain data:
- xdm.source.ipv4
- xdm.source.user.upn
- xdm.event.original_event_type
- xdm.event.outcome
- xdm.event.outcome_reason
- xdm.event.operation

p.s 
you can talk with Eido Epstain as he very familiar with it.

File: Packs/IBMSecurityVerify/ModelingRules/IBMSecurityVerify/IBMSecurityVerify.xif [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+    xdm.auth.service = data -> subtype,
+    xdm.source.user.identifier = data -> userid,
+    xdm.source.user.username = coalesce(src_user, data -> username),
+    xdm.source.user.upn = if(src_user != null, data -> username),
+    xdm.target.application.name = data -> applicationname,
```
Comment (yassorpanw) [inline Packs/IBMSecurityVerify/ModelingRules/IBMSecurityVerify/IBMSecurityVerify.xif:61]: we put the accessed app under xdm.target.resource.name

File: Packs/IBMSecurityVerify/ModelingRules/IBMSecurityVerify/IBMSecurityVerify.xif [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+    xdm.source.user.upn = if(src_user2 != null, data -> username ,src_user1 != null, data -> performedby_username),
+    xdm.target.resource.id = data -> entitlementid,
+    xdm.target.resource.name = data -> entitlementname,
+    xdm.target.resource.sub_type = data -> entitlementsubtype,
+    xdm.target.resource.type = if(data -> entitlementtype != null AND data -> entitlementcategory != null, concat(data -> entitlementtype, " - ", data -> entitlementcategory), coalesce(data -> entitlementtype, data -> entitlementcategory, data -> resource )), // The "type" field represents the role, while the "category" field specifies the role type, e.g., "role - admin."
+    xdm.source.user.groups = to_json_string(data -> entitlements) -> []
+| call generic_fields_mapping;
+
+/*
+Mapping of 'sso' event types.
```
Comment (yassorpanw) [inline Packs/IBMSecurityVerify/ModelingRules/IBMSecurityVerify/IBMSecurityVerify.xif:51]: The following fields are missing:
- xdm.event.operation.sub.type = Generic SSO
- xdm_target_resource_id (data.applicationid)
- xdm.source.user.username (similar to DUO https://github.com/demisto/content/pull/29139/files)
- xdm.source.user.first.name (similar to DUO)
- xdm.source.user.last.name (similar to DUO)

File: Packs/IBMSecurityVerify/ModelingRules/IBMSecurityVerify/IBMSecurityVerify.xif [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+| call generic_fields_mapping;
+
+/*
+Mapping of 'authentication' event types
+*/
+filter event_type = "authentication"
+| alter
+    src_user = arrayindex(regextract(data -> username, "(.+)\@") ,0) // The username can sometimes appear as a username and other times as an email.
+| alter
+    xdm.event.operation_sub_type = coalesce(data -> dict_op, data -> action),
```
Comment (yassorpanw) [inline Packs/IBMSecurityVerify/ModelingRules/IBMSecurityVerify/IBMSecurityVerify.xif:72]: For xdm.event.operation_sub_type we aim to normalize into one of the following groups:
- voice
- sms
- email
- hardware_token 
- trusted_login
- applications
see examples here: https://gitlab.xdr.pan.local/xdr/analytics-research/analytics_content/-/merge_requests/3284/diffs#5654a978f9cf5fd2b142313a3f8bd61ace085a96

so in our case we can do something like (this is not the full mapping, need to dive deeper):
case
           when data.subtype in ("user_password") then "password"
           when data.mfamethod in ("FIDO2") then "hardware_token"
           when data.mfamethod in ("Email OTP") then "email"
           when data.subtype in ("saml") then "Generic SSO"
end  as xdm.event.operation_sub_type

Comment (yassorpanw) [reply Packs/IBMSecurityVerify/ModelingRules/IBMSecurityVerify/IBMSecurityVerify.xif:72]: Need to be normalized like in Okta (https://gitlab.xdr.pan.local/xdr/analytics-research/analytics_content/-/merge_requests/3284/diffs#5f8e4b344d259c9bf1c1760dc8d2087687aa5113)
        # Email OTP: A one-time passcode sent via email.
        # SMS OTP: A one-time passcode sent via SMS to the user’s phone.
        # Voice callback OTP: A code delivered via a voice call to the user’s phone.
        # Time-based OTP (TOTP): A code generated by an authenticator app (such as Google Authenticator or IBM Verify).
        # IBM Verify App (user presence): A push notification sent to the IBM Verify mobile app that requires user confirmation.
        # IBM Verify App (biometric): A biometric challenge, such as fingerprint or facial recognition, via the IBM Verify app.
        # FIDO2: A hardware security key or biometric method compliant with the FIDO2 standard.

File: Packs/IBMSecurityVerify/ModelingRules/IBMSecurityVerify/IBMSecurityVerify.xif [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+    xdm.event.description = data -> cause,
+    xdm.auth.service = data -> sourcetype,
+    xdm.source.user.identifier = if(data -> subject = "UNKNOWN", null, coalesce(data -> subject, data -> performedby)),
+    xdm.target.url = data -> ["target"],
+    xdm.source.user.username = coalesce(src_user, data -> username),
+    xdm.source.user.upn = if(src_user != null, data -> username),
+    xdm.source.host.device_id = data -> deviceid,
+    xdm.auth.mfa.method = data -> mfamethod,
+    xdm.auth.mfa.client_details = data -> mfadevice,
```
Comment (yassorpanw) [inline Packs/IBMSecurityVerify/ModelingRules/IBMSecurityVerify/IBMSecurityVerify.xif:81]: I don't think that this mapping is correct as the data -> mfadevice contains (only in MFA auth events) the name of the second factor (it's a pretty name that the user submits), so i think it may me confusing

File: Packs/IBMSecurityVerify/ModelingRules/IBMSecurityVerify/IBMSecurityVerify.xif [hunk 6]
```diff
@@ -0,0 +1,139 @@
...
+
+/*
+Mapping of 'authentication' event types
+*/
+filter event_type = "authentication"
+| alter
+    src_user = arrayindex(regextract(data -> username, "(.+)\@") ,0) // The username can sometimes appear as a username and other times as an email.
+| alter
+    xdm.event.operation_sub_type = coalesce(data -> dict_op, data -> action),
+    xdm.event.description = data -> cause,
```
Comment (yassorpanw) [inline Packs/IBMSecurityVerify/ModelingRules/IBMSecurityVerify/IBMSecurityVerify.xif:73]: we want to map the data -> cause into xdm.event.outcome_reason, here also we want to normalize it groups (like in the example above of xdm.event.operation_sub_type). In this case we're trying to stick with the following groups:
- user_does_not_exist
- mfa_failure
- mfa_expired
- user_reject
- user_cancelled
- failed_login
- bad_credentials

File: Packs/IBMSecurityVerify/ModelingRules/IBMSecurityVerify/IBMSecurityVerify.xif [hunk 7]
```diff
@@ -0,0 +1,139 @@
...
+    xdm.target.resource.name = data -> entitlementname,
+    xdm.target.resource.sub_type = data -> entitlementsubtype,
+    xdm.target.resource.type = if(data -> entitlementtype != null AND data -> entitlementcategory != null, concat(data -> entitlementtype, " - ", data -> entitlementcategory), coalesce(data -> entitlementtype, data -> entitlementcategory, data -> resource )), // The "type" field represents the role, while the "category" field specifies the role type, e.g., "role - admin."
+    xdm.source.user.groups = to_json_string(data -> entitlements) -> []
+| call generic_fields_mapping;
+
+/*
+Mapping of 'sso' event types.
+*/
+filter event_type = "sso"
```
Comment (yassorpanw) [inline Packs/IBMSecurityVerify/ModelingRules/IBMSecurityVerify/IBMSecurityVerify.xif:53]: In 102 events the xdm.event.outcome should be only XDM_CONST.OUTCOME_SUCCESS or XDM_CONST.OUTCOME_FAILED

File: Packs/IBMSecurityVerify/ModelingRules/IBMSecurityVerify/IBMSecurityVerify.xif [hunk 8]
```diff
@@ -0,0 +1,139 @@
...
+    xdm.source.user.username = coalesce(src_user, data -> username),
+    xdm.source.user.upn = if(src_user != null, data -> username),
+    xdm.target.application.name = data -> applicationname,
+    xdm.source.host.device_id = data -> deviceid //valid only if its an Android or iPhone device.
+| call generic_fields_mapping;
+
+/*
+Mapping of 'authentication' event types
+*/
+filter event_type = "authentication"
```
Comment (yassorpanw) [inline Packs/IBMSecurityVerify/ModelingRules/IBMSecurityVerify/IBMSecurityVerify.xif:68]: In 102 events the xdm.event.outcome should be only XDM_CONST.OUTCOME_SUCCESS or XDM_CONST.OUTCOME_FAILED

File: Packs/IBMSecurityVerify/ModelingRules/IBMSecurityVerify/IBMSecurityVerify.xif [hunk 9]
```diff
@@ -0,0 +1,139 @@
...
+| alter
+    src_user = arrayindex(regextract(data -> username, "(.+)\@") ,0) // The username can sometimes appear as a username and other times as an email.
+| alter
+    xdm.event.operation_sub_type = coalesce(data -> dict_op, data -> action),
+    xdm.event.description = data -> cause,
+    xdm.auth.service = data -> sourcetype,
```
Comment (yassorpanw) [inline Packs/IBMSecurityVerify/ModelingRules/IBMSecurityVerify/IBMSecurityVerify.xif:74]: For the xdm.auth.service, we aim to determine whether authentication failures occur on the IDP or SP side. For example, if a user fails due to a wrong password, it's likely an IDP issue, whereas failures due to lack of permissions may be caused by the SP. Proper mapping of these errors requires a thorough understanding of the entire session flow from the data source.

File: Packs/IBMSecurityVerify/ModelingRules/IBMSecurityVerify/IBMSecurityVerify.xif [hunk 10]
```diff
@@ -0,0 +1,139 @@
...
+    xdm.source.user.identifier = if(data -> subject = "UNKNOWN", null, coalesce(data -> subject, data -> performedby)),
```
Comment (yassorpanw) [inline Packs/IBMSecurityVerify/ModelingRules/IBMSecurityVerify/IBMSecurityVerify.xif:75]: I think this should be coalesce(data.userid, data.subject) or even only data.userid - need to dive a bit deeper

File: Packs/IBMSecurityVerify/ModelingRules/IBMSecurityVerify/IBMSecurityVerify.xif [hunk 11]
```diff
@@ -0,0 +1,139 @@
...
+    xdm.target.url = data -> ["target"],
+    xdm.source.user.username = coalesce(src_user, data -> username),
+    xdm.source.user.upn = if(src_user != null, data -> username),
+    xdm.source.host.device_id = data -> deviceid,
+    xdm.auth.mfa.method = data -> mfamethod,
+    xdm.auth.mfa.client_details = data -> mfadevice,
+    xdm.event.original_event_type = coalesce(data -> messageid, data -> messageId),
```
Comment (yassorpanw) [inline Packs/IBMSecurityVerify/ModelingRules/IBMSecurityVerify/IBMSecurityVerify.xif:82]: xdm.event.original_event_type = event_type

File: Packs/IBMSecurityVerify/ModelingRules/IBMSecurityVerify/IBMSecurityVerify.xif [hunk 12]
```diff
@@ -0,0 +1,139 @@
...
+    xdm.session_context_id = data -> usersessionid
```
Comment (yassorpanw) [inline Packs/IBMSecurityVerify/ModelingRules/IBMSecurityVerify/IBMSecurityVerify.xif:83]: I think that correlationid is better mapping for that - we should understand what is the best field that describes a "session" in order to correlate several events into a session

File: Packs/IBMSecurityVerify/ModelingRules/IBMSecurityVerify/IBMSecurityVerify.xif [hunk 13]
```diff
@@ -0,0 +1,139 @@
...
+    xdm.auth.service = data -> subtype,
+    xdm.source.user.identifier = data -> userid,
+    xdm.source.user.username = coalesce(src_user, data -> username),
+    xdm.source.user.upn = if(src_user != null, data -> username),
+    xdm.target.application.name = data -> applicationname,
+    xdm.source.host.device_id = data -> deviceid //valid only if its an Android or iPhone device.
+| call generic_fields_mapping;
+
+/*
+Mapping of 'authentication' event types
```
Comment (yassorpanw) [inline Packs/IBMSecurityVerify/ModelingRules/IBMSecurityVerify/IBMSecurityVerify.xif:66]: The following fields are missing:

xdm_target_resource_id (data.applicationid)
xdm.source.user.username (similar to DUO https://github.com/demisto/content/pull/29139/files)
xdm.source.user.first.name (similar to DUO)
xdm.source.user.last.name (similar to DUO)
xdm.event.original_event_type = event_type

File: Packs/IBMSecurityVerify/ModelingRules/IBMSecurityVerify/IBMSecurityVerify.xif (hunk 1 above)
Comment (sharonfi99) [reply Packs/IBMSecurityVerify/ModelingRules/IBMSecurityVerify/IBMSecurityVerify.xif:56]: A couple of questions:

1. Regarding the 5 tuples, there is no information about whether it's TCP/UDP. Please specify the field from the XDM schema where you want this to be populated. I found only application protocol-related fields.
2. In `xdm.event.original_event_type`, it's currently populated with `messageid`, a unique ID number for the message in data.cause. This ID refers to a specific event subtype, in this case, authentication. 
3. What should be populated in `xdm.event.outcome_reason`?
4. Since `xdm.event.operation` is an ENUM and we don't have all the possible event subtypes documented in the IBM documentation, I've populated `xdm.event.operation_sub_type` for the 'authentication' event type. However, for 'sso' events, we don't have any additional information that can be extracted from the logs.

File: Packs/IBMSecurityVerify/ModelingRules/IBMSecurityVerify/IBMSecurityVerify.xif (hunk 2 above)
Comment (sharonfi99) [reply Packs/IBMSecurityVerify/ModelingRules/IBMSecurityVerify/IBMSecurityVerify.xif:61]: If we change this, the data regarding the entitlements given to the user won't be mapped.

File: Packs/IBMSecurityVerify/ModelingRules/IBMSecurityVerify/IBMSecurityVerify.xif [hunk 14]
```diff
@@ -0,0 +1,139 @@
...
+filter event_type = "sso"
+| alter
+    src_user = arrayindex(regextract(data -> username, "(.+)\@") ,0) // The username can sometimes appear as a username and other times as an email.
+| alter
+    xdm.auth.service = data -> subtype,
+    xdm.source.user.identifier = data -> userid,
+    xdm.source.user.username = coalesce(src_user, data -> username),
+    xdm.source.user.upn = if(src_user != null, data -> username),
+    xdm.target.application.name = data -> applicationname,
+    xdm.source.host.device_id = data -> deviceid //valid only if its an Android or iPhone device.
```
[the remark this answers is not in the document]
Comment (sharonfi99) [reply Packs/IBMSecurityVerify/ModelingRules/IBMSecurityVerify/IBMSecurityVerify.xif:62]: We followed the documentation in this case, and there were no smartphone-related events generated by the lab.

File: Packs/IBMSecurityVerify/ModelingRules/IBMSecurityVerify/IBMSecurityVerify.xif (hunk 3 above)
Comment (sharonfi99) [reply Packs/IBMSecurityVerify/ModelingRules/IBMSecurityVerify/IBMSecurityVerify.xif:51]: For 'xdm.source.user.first.name' and 'xdm.source.user.last.name', based on the DUO documentation, the logs contain fields like 'sam.smith@email.com', so It makes sense to extract first and last names, but in IBM we're receiving UPN or username.

File: Packs/IBMSecurityVerify/ModelingRules/IBMSecurityVerify/IBMSecurityVerify.xif (hunk 13 above)
Comment (sharonfi99) [reply Packs/IBMSecurityVerify/ModelingRules/IBMSecurityVerify/IBMSecurityVerify.xif:66]: In 'authentication' events there isn't `data.applcationid` / `data.applicationname`
`xdm.source.user.username `- is being extracted and populated. 
For `xdm.source.user.first.name`and `xdm.source.user.last.name`, based on the DUO documentation, the logs contain fields like '[sam.smith@email.com](mailto:sam.smith@email.com)', so It makes sense to extract first and last names, but in IBM we're receiving UPN or username.
In `xdm.event.original_event_type`, it's currently populated with messageid, a unique ID number for the message in data.cause. This ID refers to a specific event subtype in an authentication event.

File: Packs/IBMSecurityVerify/ModelingRules/IBMSecurityVerify/IBMSecurityVerify.xif (hunk 10 above)
Comment (sharonfi99) [reply Packs/IBMSecurityVerify/ModelingRules/IBMSecurityVerify/IBMSecurityVerify.xif:75]: For 'authentication' events,  the `data.userid` field is not present.
In this logic `if(data -> subject = "UNKNOWN", null, coalesce(data -> subject, data -> performedby))`
If `data.subject`  contains 'UNKNOWN' we populate the field with a null value.
Based on the logs,  there is a correlation between `data.subject` and `data.performedby`, which mostly contain the same values.

File: Packs/IBMSecurityVerify/ModelingRules/IBMSecurityVerify/IBMSecurityVerify.xif (hunk 11 above)
Comment (sharonfi99) [reply Packs/IBMSecurityVerify/ModelingRules/IBMSecurityVerify/IBMSecurityVerify.xif:82]: In xdm.event.original_event_type, it's currently populated with messageid, a unique ID number for the message in data.cause. This ID refers to a specific event subtype in an authentication event.

File: Packs/IBMSecurityVerify/ModelingRules/IBMSecurityVerify/IBMSecurityVerify.xif (hunk 3 above)
Comment (yasta5) [reply Packs/IBMSecurityVerify/ModelingRules/IBMSecurityVerify/IBMSecurityVerify.xif:51]: @yassorpanw As discussed, we can resolve this converstaion:

1. xdm_target_resource_id - Added (data.applicationid).
2. xdm.source.user.username - Added
3. xdm.source.user.first.name - Can’t be achieved (username can be anything)
4. xdm.source.user.last.name - Can’t be achieved (username can be anything)
5. xdm.event.original_event_type - Added

File: Packs/IBMSecurityVerify/ModelingRules/IBMSecurityVerify/IBMSecurityVerify.xif (hunk 1 above)
Comment (yasta5) [reply Packs/IBMSecurityVerify/ModelingRules/IBMSecurityVerify/IBMSecurityVerify.xif:57]: @yassorpanw 

1. xdm.event.tags - Verified
2. xdm.source.user.upn - Verified
3. xdm.event.type = "authentication" - Verified
4. xdm.event.original_event_type - Verified
5. xdm.event.outcome - Verified
6. xdm.event.operation (XDM_CONST.OPERATION_TYPE_AUTH_LOGIN) - Added
7. xdm.event.outcome_reason - Waiting for more information / assistance from IBM.
8. 5 Tuples - Verified and added except for **xdm.network.ip_protocol** as discussed. I also didn't see that field populated in either Duo or Okta.

To summarize, there are 2 open things:

1. xdm.event.outcome_reason
2. 5 Tuples - Your response about what should we do regarding the **xdm.network.ip_protocol**.

File: Packs/IBMSecurityVerify/ModelingRules/IBMSecurityVerify/IBMSecurityVerify.xif (hunk 13 above)
Comment (yasta5) [reply Packs/IBMSecurityVerify/ModelingRules/IBMSecurityVerify/IBMSecurityVerify.xif:77]: @yassorpanw As discussed, we can resolve this conversation:

1. xdm.target.resource.id - No application fields in authentication events.
2. xdm.source.user.username - Added
3. xdm.source.user.first.name - Can’t be achieved (username can be anything)
4. xdm.source.user.last.name - Can’t be achieved (username can be anything)
5. xdm.event.original_event_type - set to event_type to help with tracing back the event.
6. xdm.event.operation - Added (if(lowercase(data.subtype) = "MFA",  XDM_CONST.OPERATION_TYPE_AUTH_MFA, XDM_CONST.OPERATION_TYPE_AUTH_LOGIN)

File: Packs/IBMSecurityVerify/ModelingRules/IBMSecurityVerify/IBMSecurityVerify.xif (hunk 10 above)
Comment (yasta5) [reply Packs/IBMSecurityVerify/ModelingRules/IBMSecurityVerify/IBMSecurityVerify.xif:89]: @yassorpanw As discussed, we can resolve this conversation:

The current mapping is correct, there is no userid in authentication logs, and based on the docs subject’s description is ”Verify user ID that caused event to be generated.”

File: Packs/IBMSecurityVerify/ModelingRules/IBMSecurityVerify/IBMSecurityVerify.xif (hunk 12 above)
Comment (yasta5) [reply Packs/IBMSecurityVerify/ModelingRules/IBMSecurityVerify/IBMSecurityVerify.xif]: @yassorpanw 
I think that we should use usersessionid field, because comparing to correlationid, there are events with the same usersessionid while correlationid is unique for every event (in sso and authentication events). 

The only downside is that usersessionid has null values for some events, but I think that this is the that should be use to correlate events under the same authentication flow.

If you agree, we can resolve this conversation.

File: Packs/IBMSecurityVerify/ModelingRules/IBMSecurityVerify/IBMSecurityVerify.xif (hunk 14 above)
[the remark this answers is not in the document]
Comment (yasta5) [reply Packs/IBMSecurityVerify/ModelingRules/IBMSecurityVerify/IBMSecurityVerify.xif]: @yassorpanw 
I think that this field isn’t just for mobile devices. I think that the note is relevant for the mdm fields. It is true that we don’t have any value for this field but the mapping is logical.

Removed the comment. 

Also, added data -> mfadevice mapping to xdm.auth.mfa.client_details field for authentication events.

File: Packs/IBMSecurityVerify/ModelingRules/IBMSecurityVerify/IBMSecurityVerify.xif (hunk 11 above)
Comment (yasta5) [reply Packs/IBMSecurityVerify/ModelingRules/IBMSecurityVerify/IBMSecurityVerify.xif]: @yassorpanw 
xdm.event.original_event_type - set to event_type to help with tracing back the event.

File: Packs/IBMSecurityVerify/ModelingRules/IBMSecurityVerify/IBMSecurityVerify.xif (hunk 6 above)
Comment (yasta5) [reply Packs/IBMSecurityVerify/ModelingRules/IBMSecurityVerify/IBMSecurityVerify.xif:92]: @yassorpanw 
I have added mapping for xdm.event.outcome_reason after I simulated events.
Here is a summary of the outcomes:

- risky_signin - yes - simulated
- bad_credentials - yes - simulated
- failed_login - yes - simulated
- user_cancelled - Not relevant, doesn't exist
- user_reject - yes - simulated
- mfa_expired - simulated but event wasn’t logged
- mfa_failure -  simulated but event wasn’t logged (only logged for deny action from the verification app)
- device_security_issues. - couldn't simulate
- auth_login_restrictions - yes - simulated
- auth_policy_access_violation - Doesn't exist in the authentication logs, only the risk logs (login with good creds generates successful login authentication event but then a risk event is generated with the action denied due to policy)
- account_locked - yes - simulated
- account_expired_or_disabled - yes - simulated
- user_does_not_exist - No logs created - simulated
- malformed_request - couldn't simulate

Please review and confirm

File: Packs/IBMSecurityVerify/ModelingRules/IBMSecurityVerify/IBMSecurityVerify.xif (hunk 9 above)
Comment (yassorpanw) [reply Packs/IBMSecurityVerify/ModelingRules/IBMSecurityVerify/IBMSecurityVerify.xif:93]: what do you say about if(
event_type = "authentication",
"IDP",
event_type = "sso",
"SP",
)

Comment (yasta5) [reply Packs/IBMSecurityVerify/ModelingRules/IBMSecurityVerify/IBMSecurityVerify.xif:93]: @yassorpanw 
I was thinking about it, but I wanted to simulate a failure of SSO to verify that there is no authentication event created without any luck.
I can implement it needed.

**Review by OBavly — APPROVED** [review]
